CMakeLists.txt

```diff
@@ -11,7 +12,7 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Debug)
 endif(NOT CMAKE_BUILD_TYPE)
 
-if (ASAN AND COMPILE_PYBIND)
+if (ASAN AND (COMPILE_BINDINGS OR BUILD_PIP_PACKAGE))
     message(WARNING "Disabling ASAN with python bindings")
     set(ASAN OFF)
 endif()
```

Comment:
It seems like you're right and adding `LD_PRELOAD=/usr/lib/libasan.so` is enough to make it working. So yes, I agree on replacing the warning instead of disabling ASAN, however, this should be done in a separate PR.